realm/src/main/java/io/realm/dynamic/DynamicRealmObject.java

```diff
@@ -0,0 +1,516 @@
+    }
+
+    /**
+     * Sets the {@code float} value of the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Value to insert.
+     * @throws IllegalArgumentException if field name doesn't exists or isn't an integer field.
+     */
+    public void setFloat(String fieldName, float value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        row.setFloat(columnIndex, value);
+    }
+
+    /**
+     * Sets the {@code double} value of the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Value to insert.
+     * @throws IllegalArgumentException if field name doesn't exists or isn't a double field.
+     */
+    public void setDouble(String fieldName, double value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        row.setDouble(columnIndex, value);
+    }
+
+    /**
+     * Sets the {@code String} value of the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Value to insert.
+     * @throws IllegalArgumentException if field name doesn't exists or isn't a String field.
+     */
+    public void setString(String fieldName, String value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        row.setString(columnIndex, value);
+    }
+
+    /**
+     * Sets the binary value of the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Value to insert.
+     * @throws IllegalArgumentException if field name doesn't exists or isn't a binary field.
+     */
+    public void setBlob(String fieldName, byte[] value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        row.setBinaryByteArray(columnIndex, value);
+    }
+
+    /**
+     * Sets the {@code Date} value of the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Value to insert.
+     * @throws IllegalArgumentException if field name doesn't exists or isn't a Date field.
+     */
+    public void setDate(String fieldName, Date value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        row.setDate(columnIndex, value);
+    }
+
+    /**
+     * Sets a reference to another object on the given field.
+     *
+     * @param fieldName Field name.
+     * @param value Object to link to.
+     * @throws IllegalArgumentException if field name doesn't exists, it doesn't link to other Realm objects, or the type
+     * of DynamicRealmObject doesn't match.
+     */
+    public void setObject(String fieldName, DynamicRealmObject value) {
+        long columnIndex = row.getColumnIndex(fieldName);
+        if (value == null) {
+            row.nullifyLink(columnIndex);
+        } else {
+            if (value.realm == null || value.row == null) {
+                throw new IllegalArgumentException("Cannot link to objects that are not part of the Realm.");
+            }
+            if (value.realm != null && !realm.getConfiguration().equals(value.realm.getConfiguration())) {
```

Comment:
Just a few lines above, an exception is thrown if `value.realm` is `null`. If we get to this line, `value.realm` cannot be `null`, right?